homeassistant/components/progettihwsw/config_flow.py

```diff
@@ -0,0 +1,102 @@
+"""Config flow for ProgettiHWSW Automation integration."""
+
+from ProgettiHWSW.ProgettiHWSWAPI import ProgettiHWSWAPI
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required("host"): str, vol.Required("port", default=80): int}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user host input."""
+
+    api_instance = ProgettiHWSWAPI(f'{data["host"]}:{data["port"]}')
+    is_valid = await api_instance.check_board()
+
+    if is_valid is False:
+        raise CannotConnect
+
+    return {
+        "title": is_valid["title"],
+        "relay_count": is_valid["relay_count"],
+        "input_count": is_valid["input_count"],
+        "is_old": is_valid["is_old"],
+    }
+
+
+async def validate_input_relay_modes(data):
+    """Validate the user input in relay modes form."""
+    for mode in data.values():
+        if mode not in ("bistable", "monostable"):
+            raise WrongInfo
+
+    return True
+
+
+class ProgettiHWSWConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for ProgettiHWSW Automation."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_relay_modes(self, user_input=None):
+        """Manage relay modes step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                await validate_input_relay_modes(user_input)
+                whole_data = user_input
+                whole_data.update(self.s1_in)
+            except WrongInfo:
+                errors["base"] = "wrong_info_relay_modes"
+            except Exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+            else:
+                return self.async_create_entry(
```

Comment:
We should check that we don't set up the same board, ie host and port, for more than one config entry, before creating the entry.